diag_manager/fms_diag_file_object.F90

```diff
@@ -751,9 +783,64 @@ subroutine open_diag_file(this, time_step)
     diag_file%next_open = diag_time_inc(diag_file%next_open, VERY_LARGE_FILE_FREQ, DIAG_DAYS)
   endif
 
-!TODO: closing the file here for now, just to see if it works
-  call close_file(diag_file%fileobj)
+  file_is_opened = .true.
+  domain => null()
+  diag_file => null()
 end subroutine open_diag_file
 
+!< @brief Writes the axis metadata for the file
+subroutine write_metadata(this, diag_axis)
```

Comment:
Is this specific for an axis?  If it is, it should be called `write_axis_metadata`